File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/ActivityFeedDownloadedEvent.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;

import com.shareyourproxy.api.domain.model.ActivityFeedItem;

import java.util.List;

/**
 * The activities feed has been downloaded.
 */
public class ActivityFeedDownloadedEvent extends EventCallback {
    private final static java.lang.ClassLoader CL =
        ActivityFeedDownloadedEvent.class.getClassLoader();
    public static final Creator<ActivityFeedDownloadedEvent> CREATOR =
        new Creator<ActivityFeedDownloadedEvent>() {
            @Override
            public ActivityFeedDownloadedEvent createFromParcel(Parcel in) {
                return new ActivityFeedDownloadedEvent(
                    (List<ActivityFeedItem>) in.readValue(CL));
            }

            @Override
            public ActivityFeedDownloadedEvent[] newArray(int size) {
                return new ActivityFeedDownloadedEvent[size];
            }
        };
    public final List<ActivityFeedItem> feedItems;

    public ActivityFeedDownloadedEvent(List<ActivityFeedItem> feedItems) {
        this.feedItems = feedItems;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(feedItems);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/ActivityFeedDownloadedEvent.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable

import com.shareyourproxy.api.domain.model.ActivityFeedItem

/**
 * The activities feed has been downloaded.
 */
internal final class ActivityFeedDownloadedEvent(val feedItems: List<ActivityFeedItem>) : EventCallback() {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(feedItems)
    }

    companion object {
        private val CL = ActivityFeedDownloadedEvent::class.java.classLoader
        val CREATOR: Parcelable.Creator<ActivityFeedDownloadedEvent> = object : Parcelable.Creator<ActivityFeedDownloadedEvent> {
            @Suppress("UNCHECKED_CAST")
            override fun createFromParcel(parcel: Parcel): ActivityFeedDownloadedEvent {
                return ActivityFeedDownloadedEvent(parcel.readValue(CL) as List<ActivityFeedItem>)
            }

            override fun newArray(size: Int): Array<ActivityFeedDownloadedEvent?> {
                return arrayOfNulls(size)
            }
        }
    }
}
